src/components/ContactsAdd.js

```diff
@@ -1,31 +1,99 @@
 import { useState } from "react"
 import { useNavigate } from "react-router-dom";
 
-function ContactsAdd(props) {
 
+function ContactsAdd(props) {
+ const initialFormState = {
+    firstName: "",
+    lastName: "",
+    street: "",
+    city: "",
+    email: "",
+    linkedin: "",
+    twitter: ""
+  }
+const [ContactData, setContactData] = useState(initialFormState)
   // setContacts and contacts must be passed as props
   // to this component so new contacts can be added to the
   // state
+  const navigate = useNavigate()
   const { setContacts, contacts } = props
+  
+  function handleChange(event) {
+    const inputName = event.target.name
+    const inputValue = event.target.value
+    const inputType = event.target.value
 
+    if(inputName === "firstName"){
+      
+      setContactData({...ContactData, firstName: inputValue})
+    }
+    if(inputName === "lastName"){
+      setContactData({...ContactData, lastName: inputValue})
+    }
+    if(inputName === "street"){
+      setContactData({...ContactData, street: inputValue})
+    }
+    if(inputName === "city"){
+      setContactData({...ContactData, city: inputValue})
+    }
+    if(inputName === "email"){
+      setContactData({...ContactData, email: inputValue})
+    }
+    if(inputName === "linkedIn"){
+      setContactData({...ContactData, linkedin: inputValue})
+    }
+    if(inputName === "twitter"){
+      setContactData({...ContactData, twitter: inputValue})
+    }
+    console.log("the constactdata",ContactData)
```

Comment:
I did this slightly differently - so I'm curious as to why you choose to setContactData for every bit of data? My approach was to create a variable adding all the values to the variable and then at the end did a single setContactData using said variable.